lib/mobility/backends/active_record/key_value.rb

```diff
@@ -155,6 +156,7 @@ def visit_default(_)
         key_column         = options[:key_column]
         value_column       = options[:value_column]
         belongs_to         = options[:belongs_to]
+        include_other_translation_classes = options[:include_other_translation_classes]
```

Comment:
```suggestion
        destroy_descendants = options[:destroy_descendants]
```

Reply:
@shioyama is probably the best person to answer those questions 🙂

Reply:
Sorry for the late reply!

> Maybe we should first clarify the reason why its necessary to take the other Translation classes other then the one inside translation_class into account while destroying?

Yes I admit this is quite hard to understand from the code.

The reason why the code attempts to destroy _all_ translation type records is that you can start by declaring (say) a `text` translation type and then later switch to `string`, and you will leave `text` translation orphans attached to the model record.

So the code is intended to guarantee that when the model is destroyed, all translations attached to it, regardless of their type, will also get destroyed.

TBH, looking at this code now (a couple years after I wrote it), I'm not sure that the complexity tradeoff is worth it. But OTOH it _is_ very easy to change translation types and leave behind orphaned translations.

Anyway, I don't really want to change the default, but I agree it should be possible to disable it. My only question is: should the option be to _completely_ disable the `after_destroy` hook, or disable destroying the descendants (as you have done here) ?

I'm a bit concerned that someone may come along later and want to implement the whole `after_destroy` thing differently, at which point we'd need yet another option for this. If that's the case, maybe we should just have one option which disables the whole thing (so no cleanup of translations of any kind) and anyone who uses it has to manage translations cleanup on their own.

WDYT?